test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -196,15 +196,22 @@ class test_uct_sockaddr : public uct_test {
         m_connect_addr.set_port(m_listen_addr.get_port());
     }
 
-    void listen_and_connect() {
-        start_listen(test_uct_sockaddr::conn_request_cb);
-
-        client_user_data *user_data = new client_user_data(*this, *m_client, 0);
-        m_client->connect_to_sockaddr(0, m_connect_addr, client_resolve_cb,
-                                      client_connect_cb, client_disconnect_cb,
-                                      user_data);
+    void connect(uct_test::entity &e, unsigned index,
+                 uct_ep_disconnect_cb_t disconnect_cb) {
+        client_user_data *user_data = new client_user_data(*this, e, index);
```

Comment:
So the point of the PR is to add lock on async to prevent race of accessing user data?
can you pls update PR and/or commit description to include this?

Reply:
done